bind/std_string.cpp

```diff
@@ -18,7 +24,7 @@ class basic_string {
 
 } CPPMM_OPAQUEBYTES CPPMM_IGNORE_UNBOUND;
 
-using string = ::std::string;
+using string = std::string;
```

Comment:
I somewhat copied this from the cppmm test suite. I'm really not sure if this is correct, but the `_GLIBCXX_USE_CXX11_ABI` changes seemed relevant as we prepare for the new ABI. Heads-up that cppmm use `#ifdef` which is incorrect because this value is allowed to be either `0` or `1` and `ifdef` only checks for existence. If that's indeed a mistake we'll patch cppmm's test case to use `#if` as well.

I wasn't sure why some of the signatures use `::std` here and some use the local `std` namespace alias which expands to the ABI-specific std. Should all of them use the local std namespace alias? If so I'll tweak that too in a follow-up commit.

Reply:
Yep the test case in cppmm itself is out of date, thanks for pointing that out. I've just pushed the correct version:
```c++
#if defined(_GLIBCXX_USE_CXX11_ABI)
#if _GLIBCXX_USE_CXX11_ABI
namespace std = ::std::__cxx11;
#else
namespace std = ::std;
#endif
#elif defined(_LIBCPP_VERSION)
namespace std = ::std::_LIBCPP_ABI_NAMESPACE;
#else
namespace std = ::std;
#endif
```

The reason some signiatures here use `::std` and some use `std` is that some are referencing the local namespace and some are referencing the global one. These are set correctly so don't change them, the above fix should make everything work correctly. If it doesn't for you let me know. I agree it's somewhat confusing - it's set this way because we need the type names in the parameters to expand to e.g. `::std::__cxx11::basic_string<char>`, but `string` is typedefed at the top-level `std` namespace.